.github/actions/send_notification/action.yml

```diff
@@ -0,0 +1,39 @@
+name: Send SNS failure notification
+description: Send an SNS notification on failure using the AWS CLI
+inputs:
+  role-to-assume:
+    description: AWS IAM role to assume when sending the notification
+    required: true
+  sns-topic-arn:
+    description: AWS SNS topic ARN to use when sending the notification
+    required: true
+runs:
+  using: composite
+  steps:
+    - name: Configure AWS credentials
+      uses: aws-actions/configure-aws-credentials@v4
+      with:
+        role-to-assume: ${{ inputs.role-to-assume }}
+        aws-region: us-east-1
+
+    - name: Create failure notification components
+      run: |
+        TIMESTAMP=$(date -u +"%Y-%m-%dT%H:%M:%S")
+        {
+          echo "NOTIFICATION_SUBJECT=dbt tests failed for workflow run: $GITHUB_RUN_ID UTC"
+          echo "NOTIFICATION_BODY<<EOF"
+          echo "dbt tests failed for workflow $GITHUB_RUN_ID, run on $TIMESTAMP"
+          echo ""
+          echo "Link to failing workflow: \
+        https://github.com/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID"
+          echo EOF
+        } >> "$GITHUB_ENV"
```

Comment:
This constructs a little multi-line string for the email body. Here's a sample output:

> dbt tests failed for workflow 6191942634, run on 2023-09-14T23:32:02Z
>
> Link to failing workflow: https://github.com/ccao-data/data-architecture/actions/runs/6191942634

Reply:
**[Suggestion, non-blocking]** Nice and simple! I think that [the `$GITHUB_OUPUT` parameter](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-output-parameter) might be a more appropriate place for these variables to live, since configuring them as output variables would allow subsequent steps to opt into reading them (in contrast to the `$GITHUB_ENV` environment variables, which all subsequent steps have to inherit, including steps in the workflow that calls this action).

Reply:
Good point! Fixed in 98134ad99655d7c9d14b44453289c8b8ffde5f37.